spec/services/person_constructor_spec.rb

```diff
@@ -59,6 +59,26 @@
 
       expect(person.location.city).to eq 'city'
     end
+
+    it "normalizes zip codes before assigning to location" do
+      params = { zip_code: '12345-0001' }
+      expected_params = { zip_code: '12345' }
+      stub_person_finder(expected_params)
+
+      person = PersonConstructor.build(params)
+
+      expect(person.location.zip_code).to eq '12345'
+    end
+
+    it "rejects bad zip codes" do
+      params = { zip_code: '1231' }
+      expected_params = { zip_code: nil }
+      stub_person_finder(expected_params)
```

Comment:
Actually, we really shouldn't be normalizing a property of another 
object in `Person`.  Let's do this normalization in `Location`.

-Mario



On 4/15/2016 8:33 PM, Jacob Turino wrote:
>
> In spec/services/person_constructor_spec.rb 
> <https://github.com/MayOneUS/mayday-2.0-backend/pull/103#discussion_r59919799>:
>
> > @@ -59,6 +59,26 @@
> >
> >        expect(person.location.city).to eq 'city'
> >      end
> > +
> > +    it "normalizes zip codes before assigning to location" do
> > +      params = { zip_code: '12345-0001' }
> > +      expected_params = { zip_code: '12345' }
> > +      stub_person_finder(expected_params)
> > +
> > +      person = PersonConstructor.build(params)
> > +
> > +      expect(person.location.zip_code).to eq '12345'
> > +    end
> > +
> > +    it "rejects bad zip codes" do
> > +      params = { zip_code: '1231' }
> > +      expected_params = { zip_code: nil }
> > +      stub_person_finder(expected_params)
>
> this test requires too much setup to test that the correct params are 
> being passed to Location. This is why I think parameter normalization 
> should be in its own class
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> <https://github.com/MayOneUS/mayday-2.0-backend/pull/103/files/e22063fd9fc5ae2778804bca355ce2a43b330d0b#r59919799>
>